scripts/sync-manifests.js

```
@@ -0,0 +1,190 @@
+  if (canonical !== rootCanonical && !canonical.startsWith(rootCanonical + '/')) {
+    throw new Error(`[sync-manifests] Path traversal detected: ${filePath}`);
+  }
+}
+
+function atomicWrite(filePath, content) {
+  const tmp = filePath + '.tmp';
+  writeFileSync(tmp, content, 'utf8');
+  try {
+    renameSync(tmp, filePath); // atomic on Linux when on same filesystem
+  } catch (e) {
+    try { unlinkSync(tmp); } catch (_) {}
```

ESLint

[error] Empty block statement.

(no-empty)